pkg/webhook/mutation/pod_mutator/pod_mutator.go

```diff
@@ -55,16 +55,32 @@ type podMutatorWebhook struct {
 	requestCounter metric.Int64Counter
 }
 
+func checkMutationRequestBase(mutationRequest *dtwebhook.MutationRequest, err error) (*dtwebhook.MutationRequest, admission.Response, bool) {
+	emptyPatch := admission.Patched("")
+	if err != nil {
+		emptyPatch.Result.Message = fmt.Sprintf("unable to inject into pod (err=%s)", err.Error())
+		log.Error(err, "building mutation request base encountered an error")
+		return nil, emptyPatch, false
+	}
+	if mutationRequest == nil {
+		emptyPatch.Result.Message = "unable to inject into pod (no error reported)"
+		log.Error(err, "building mutation request base delivered nil result")
+		return nil, emptyPatch, false
+	}
+	return mutationRequest, emptyPatch, true
+}
```

Comment:
this function is a doozy:
- has 3 return values, already a bad sign 😬
-  feels very much like the `silentErrorResponse`, but different ?

so AFAIK`mutationRequest` is only `nil` in case there was an error, so `mutationRequest == nil` will never happen, which makes this function rather unnecessary

is there something I am not seeing ? (if so then I think its more important to make sure that `mutationRequest` is only `nil` in case of error, because it wouldn't make sense otherwise)